### PR TITLE
Add multithreading for tests

### LINKED ISSUE #120: [IDEA] Multithread Test
**What is your feature about ?**
Do the test in parallel to improve testing time ! 

**Describe what you'd like**
One thread per filter test

### PR DESCRIPTION
**Pull request by NS4nti**
# :sparkles: Pull Request Template
:bangbang: Once all the **checklist** is **done** you have to:
  * **stash merge** this pull request
  * **delete** the corresponding **branch**
  * **close** the associated **issue**

## :page_with_curl: Type of change

**New feature**: non-breaking change which adds functionality.

## :bulb: Related Issue(s)

- Resolve #120 

## :black_nib: Description

Add multi threading for tests. Tests now run in parallel, for a better time optimization.

### Ubuntu 19.04
- Python 3

## :heavy_check_mark: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

</br>

- [x] :raising_hand: **I certify on my honor that all the information provided is true, and I've done all I can to deliver a high quality code**

### CODE REVIEW

**Comment by NS4nti** [comment]
Cara bout log in the same file, we don't know who log it

**Comment by NS4nti** [comment]
> Care bout log in the same file, we don't know who log it

Uniformed logs added

File: tests/tools/output.py [hunk 1]
```diff
@@ -7,9 +7,8 @@ def print_results(results):
             print("ERROR | " + i[0])
 
 def print_result(name, function):
-    print(name + "... ", end='', flush=True)
     result = function()
     if result is True:
-        print("\33[32m OK \33[0m")
+        print(name + " :" + "\33[32m OK \33[0m", flush=True)
```
Comment (HugoSoszynski) [inline tests/tools/output.py]:
```suggestion
        print(name + ": " + "\33[32m OK \33[0m", flush=True)
```

File: tests/tools/output.py [hunk 2]
```diff
@@ -7,9 +7,8 @@ def print_results(results):
...
     else:
-        print("\033[91m ERROR \33[0m")
+        print(name + " :" + "\033[91m ERROR \33[0m", flush=True)
```
Comment (HugoSoszynski) [inline tests/tools/output.py]:
```suggestion
        print(name + ": " + "\033[91m ERROR \33[0m", flush=True)
```

**Comment by HugoSoszynski** [comment]
Closing because branch is obsolete

**Pull request closed without merging**